presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java

```diff
@@ -683,6 +683,59 @@ public JoinNode join(
         return new JoinNode(idAllocator.getNextId(), type, left, right, criteria, outputVariables, filter.map(OriginalExpressionUtils::castToRowExpression), leftHashVariable, rightHashVariable, distributionType);
     }
 
+    public JoinNode joinWithRowExpression(JoinNode.Type joinType, PlanNode left, PlanNode right, JoinNode.EquiJoinClause... criteria)
+    {
+        return joinWithRowExpression(joinType, left, right, Optional.empty(), criteria);
+    }
+
+    public JoinNode joinWithRowExpression(JoinNode.Type joinType, PlanNode left, PlanNode right, RowExpression filter, JoinNode.EquiJoinClause... criteria)
```

Comment:
just call these functions `join`

Reply:
Probably just change the existing `join` method signature to `RowExpression` and if any old method is called with expression, use `castToRowExpression` to cast into `RowExpression`. Do it in a separate commit.